packages/npm-dependencies-validation/src/utils/npmUtil.ts

```diff
@@ -1,23 +1,46 @@
 import { spawn } from "child_process";
+import { NpmCommandConfig, VscodeOutputChannel } from "../types";
 
 export function getNPM(): string {
   return /^win/.test(process.platform) ? "npm.cmd" : "npm";
 }
 
 export function invokeNPMCommand<T>(
-  commandArgs: string[],
-  cwd: string
+  config: NpmCommandConfig,
+  outputChannel?: VscodeOutputChannel
 ): Promise<T> {
+  const { commandArgs, cwd } = config;
   return new Promise((resolve, reject) => {
     const command = spawn(getNPM(), [...commandArgs, "--json"], { cwd });
 
     command.stdout.on("data", (data) => {
+      showData(`${data}`, outputChannel);
       const jsonObjResult: T = JSON.parse(data);
       resolve(jsonObjResult);
     });
 
     command.on("error", (error) => {
+      showError(error, outputChannel);
       reject(error);
     });
   });
 }
+
+function showError(error: Error, outputChannel?: VscodeOutputChannel): void {
+  const { stack } = error;
+  console.error(stack);
```

Comment:
**duplicating** console.log/error calls and sending to vscode outputChannel 
is another example why we should extract the logger interface.

We naturally do not want to always log to the console...

Reply:
Done